nomos-sdp/src/state.rs

```diff
@@ -265,6 +265,24 @@ impl From<WithdrawnState> for ProviderState {
     }
 }
 
+impl From<ProviderInfo> for ActiveState {
```

Comment:
Where are these conversions used?

Reply:
When I removed pub field from State, I need a way to initialize the state for tests. So it was either this or a constructor methods. Does that make sense?

Example:
```
 let updates = vec![FinalizedBlockEventUpdate {
            service_type,
            provider_id: provider_info_1.provider_id,
            state: ProviderState::Active(provider_info_1.into()),
            locators: BTreeSet::from_iter(declaration_update_1.locators.clone()),
        }];
        
```

Reply:
Lets add the `#[cfg(test)]` to each of them, these should not be allowed to be used outside the tests. I'll think how to remove them completely in the refactor, will probably make `state.rs` private module to the `nomos-sdp` and expose new enum similar to this `enum State { Active, Inactive, Withdrawn }`

Reply:
I think enum makes sense. I wanted to change it, but did not want to expand the scope of the PR any more. I will add the `#[cfg(test)]`

Reply:
Actually, I dont even need this, I will just use it the same way as in the tests of the state.

Reply:
Done